admin/app/controllers/admin/ReaderRevenueAdminController.scala

```diff
@@ -26,45 +26,61 @@ class ReaderRevenueAdminController(wsClient: WSClient, val controllerComponents:
+      val message = s"${bannerType.name} banner in ${region.name} redeploy by $requester at ${time.toString}}"
 
       val result = for {
-        _ <- updateContributionsBannerDeployLog(region, jsonLog.toString)
-        _ <- purgeDeployLogCache(region)
-      } yield bannerRedeploySuccessful(message, region)
+        _ <- updateBannerDeployLog(region, jsonLog.toString, bannerType)
+        _ <- purgeDeployLogCache(region, bannerType)
+      } yield bannerRedeploySuccessful(message, region, bannerType)
 
-      result.recover { case e => redeployFailed(e)}
+      result.recover { case e => redeployFailed(e, bannerType) }
     }
 
   }
 
-  private def updateContributionsBannerDeployLog(region: ReaderRevenueRegion, bannerDeployLogJson: String): Future[Unit] = {
+  private def updateBannerDeployLog(region: ReaderRevenueRegion, bannerDeployLogJson: String, banner: BannerType): Future[Unit] = {
+    log.info(s"updateBannerDeployLog $banner $region")
     val defaultJsonEncoding: String = "application/json;charset=utf-8"
-    val bucketKey = ReaderRevenueRegion.getBucketKey(region)
+    val bucketKey = ReaderRevenueRegion.getBucketKey(region, banner)
     Future(S3.putPublic(bucketKey, bannerDeployLogJson, defaultJsonEncoding))
   }
 
-  private def purgeDeployLogCache(region: ReaderRevenueRegion): Future[String] = {
-    val path = "/reader-revenue/contributions-banner-deploy-log/" + region.name
+  private def purgeDeployLogCache(region: ReaderRevenueRegion, bannerType: BannerType): Future[String] = {
+    val path = s"/reader-revenue${bannerType.path}/${region.name}"
+
     CdnPurge.soft(wsClient, DigestUtils.md5Hex(path), AjaxHost)
   }
 
-  private def bannerRedeploySuccessful(message: String, region: ReaderRevenueRegion): Result = {
-    log.info(s"$message: SUCCESSFUL")
-    Redirect(routes.ReaderRevenueAdminController.renderContributionsBannerAdmin()).flashing(
-      "success" -> (s"Banner redeployed in ${region.name}"))
+  private def getRoute(bannerType: BannerType): Call = {
+     bannerType match {
+      case ContributionsBanner => routes.ReaderRevenueAdminController.renderContributionsBannerAdmin()
+      case SubscriptionsBanner => routes.ReaderRevenueAdminController.renderSubscriptionsBannerAdmin()
+    }
   }
 
-  private def redeployFailed(error: Throwable): Result = {
-    log.error("Contributions banner redeploy FAILED", error)
-    Redirect(routes.ReaderRevenueAdminController.renderContributionsBannerAdmin()).flashing(
-    "error" -> ("Banner not redeployed"))
+  private def bannerRedeploySuccessful(message: String, region: ReaderRevenueRegion, bannerType: BannerType): Result = {
+    log.info(s"$message: SUCCESSFUL")
+
+    Redirect(getRoute(bannerType)).flashing("success" -> s"${bannerType.name} redeployed in ${region.name}")
+
   }
 
+  private def redeployFailed(error: Throwable, bannerType: BannerType): Result = {
+    log.error(s"${bannerType.name} banner redeploy FAILED", error)
```

Comment:
In here you're printing the name, in other places you're calling `toString`. You want to be consistent here.